src/CQRS/LeanCode.CQRS.AspNetCore/Registration/CQRSObjectsRegistrationSource.cs

```diff
@@ -57,9 +61,10 @@ private IEnumerable<CQRSObjectMetadata> FindCQRSObjects()
             );
 
             resultObjects.Add(metadata);
+            Objects = Objects.Append(metadata);
```

Comment:
1. This looks like a perf nighmare - new collection per _every_ added item.
2. You still can add the same object twice.

I'd have a mutable hashset here (with proper equality comparer) and ignore the fact that `Objects` collection can be silently updated. Or just use `ImmutableHashSet`  .